app/services/cached_appeal_service.rb

```diff
@@ -272,4 +277,69 @@ def veteran_names_for_correspondent_ids(correspondent_ids)
       [corr.stafkey, "#{corr.snamel.split(' ').last}, #{corr.snamef}"]
     end.to_h
   end
+
+  def appeal_ids_with_schedule_hearing_tasks
+    @appeal_ids_with_schedule_hearing_tasks ||= []
+  end
+
+  # get all ids of appeals which have open ScheduleHearingTasks
+  def filter_appeals_by_schedule_hearing_tasks(appeal_ids, appeal_type)
+    appeal_ids_with_schedule_hearing_tasks << Task.open
+      .where(appeal_id: appeal_ids, type: ScheduleHearingTask.name, appeal_type: appeal_type)
+      .pluck(:appeal_id)
+      .uniq
+    appeal_ids_with_schedule_hearing_tasks.flatten!
+  end
```

Comment:
3 suggestions. 
* rename method for clarity
* shorten Task query
* more efficient array appending: https://stackoverflow.com/questions/1801516/how-do-you-add-an-array-to-another-array-in-ruby-and-not-end-up-with-a-multi-dim
```suggestion
  def append_appeals_with_open_schedule_hearing_tasks(appeal_ids, appeal_type)
    appeal_ids_with_open_sched_hearing_task = ScheduleHearingTask.open
      .where(appeal_id: appeal_ids, appeal_type: appeal_type)
      .pluck(:appeal_id)
      .uniq
    appeal_ids_with_schedule_hearing_tasks.push(*appeal_ids_with_open_sched_hearing_task)
  end
```